packages/paste-core/layout/stack/src/index.tsx

```diff
@@ -81,37 +76,42 @@ export const getStackChildMargins = (orientation: StackOrientation, spacing: Spa
   return styles;
 };
 
-const Stack = React.forwardRef<HTMLDivElement, StackProps>(({children, orientation, spacing, ...props}, ref) => {
-  const [childrenCount, validChildren] = React.useMemo(() => {
-    const filteredChildren = React.Children.toArray(children).filter(
-      (child) => React.isValidElement(child) || typeof child === 'string'
+const Stack = React.forwardRef<HTMLDivElement, StackProps>(
+  ({children, orientation, spacing, element = 'STACK', ...props}, ref) => {
+    const [childrenCount, validChildren] = React.useMemo(() => {
+      const filteredChildren = React.Children.toArray(children).filter(
+        (child) => React.isValidElement(child) || typeof child === 'string'
+      );
+      return [filteredChildren.length, filteredChildren];
+    }, [children]);
+    const stackStyles = React.useMemo(() => getStackStyles(orientation), [orientation]);
+    const childMargins = React.useMemo(() => getStackChildMargins(orientation, spacing), [orientation, spacing]);
+    const keySeed = useUIDSeed();
+
+    return (
+      <Box element={element} {...safelySpreadBoxProps(props)} {...stackStyles} ref={ref}>
+        {validChildren.map((child, index) => {
+          return (
+            <Box
+              element={`${element}_CHILD`}
```

Comment:
I'm not sure I understand why we're giving the children an `element` prop. Wouldn't all of the children have the same name as the parent `element` and therefore all be styled the same?

Reply:
It creates an element name for the child based on the parent element name, but not exactly the same.

So yes, child elements have the same name, but it is different to the parent. It's potentially easier to see here in the tests https://github.com/twilio-labs/paste/pull/1722/files#diff-160e3be247ff4ca5d35a10f9e526e4cf0aa098b6b43b134a322e93e3c2ab75a4R159